lib/puppet/ssl/state_machine.rb

```diff
@@ -190,15 +191,15 @@ def next_state
                              {certname: Puppet[:certname], message: e.response.message})
       end
 
-      Wait.new(@ssl_context)
+      Wait.new(@machine, @ssl_context)
     end
   end
 
   # We cannot make progress, so wait if allowed to do so, or error.
   #
   class Wait < SSLState
     def next_state
-      time = Puppet[:onetime] ? 0 : Puppet[:waitforcert]
+      time = @machine.onetime ? 0 : @machine.waitforcert
```

Comment:
I thinking I should use `onetime` here, as that's really just an `puppet agent` thing (whether to apply a single catalog and stop, or continue running every X minutes). It just so happens that when using `onetime`, it implies `waitforcert = 0`

Reply:
I'll make a note of this in a follow up ticket.